src/main/java/org/snomed/aag/data/services/CriteriaItemService.java

```diff
@@ -175,17 +171,17 @@ public void verifyManual(CriteriaItem criteriaItem, boolean expectedManual) {
 	}
 
 	/**
-	 * Remove entry from collection. If entry does not have a corresponding flag in Branch metadata
-	 * or the corresponding flag in Branch metadata is false, then remove entry from collection.
+	 * Return entries in store matching enabledByFlag and authoringLevel fields.
 	 *
-	 * @param criteriaItems Collection to process
-	 * @param branch        Branch to cross reference
+	 * @param authorFlags    Field to match in query.
+	 * @param authoringLevel Field to match in query.
+	 * @return Entries in store matching enabledByFlag and authoringLevel fields.
 	 */
-	public void removeNonEnabled(Set<CriteriaItem> criteriaItems, Branch branch) {
-		verifyParams(criteriaItems, branch);
-		Set<String> authorFlags = MetadataUtil.getEnabledAuthorFlags(branch);
-		if (!authorFlags.isEmpty()) {
-			criteriaItems.removeIf(item -> item.getEnabledByFlag().stream().noneMatch(authorFlags::contains));
+	public List<CriteriaItem> findAllByEnabledByFlagInAndAuthoringLevel(Set<String> authorFlags, AuthoringLevel authoringLevel) {
+		if (authorFlags == null || authorFlags.isEmpty() || authoringLevel == null) {
+			return Collections.emptyList();
 		}
+
+		return repository.findAllByEnabledByFlagInAndAuthoringLevel(authorFlags, authoringLevel);
```

Comment:
This method will find all CriteriaItems with a matching authorFlag. I needed the authoringLevel as without it, it was hard to distinguish between project/task level items.